api/handlers/alertmanager.go

```diff
@@ -21,20 +21,23 @@ func UpdateAlertCredentials(service domain.AlertmanagerService, logger *zap.Logg
 			w.Write([]byte("invalid json body"))
 			return
 		}
-		teamName := params["teamName"]
-		alertCredential.TeamName = teamName
-		validators := []func(*domain.AlertCredential) error{
-			validateWebhooks,
-			validateEntity,
-			validatePagerdutyKey,
-		}
-		for _, v := range validators {
-			if err := v(&alertCredential); err != nil {
-				w.WriteHeader(400)
-				w.Write([]byte(err.Error()))
+		v := validator.New()
+		_ = v.RegisterValidation("webhookChecker", func(fl validator.FieldLevel) bool {
+			_, err := url.Parse(fl.Field().Interface().(string))
+			return err == nil
+		})
+		err = v.Struct(alertCredential)
+		if err != nil {
+			if _, ok := err.(*validator.InvalidValidationError); ok {
+				logger.Error("invalid validation error")
+				internalServerError(w, err, logger)
```

Comment:
If the webhook URL is incorrect we shouldn't be throwing internal server error. Shouldn't it be bad request or something?